internal/controller/projectdevelopmentstream_controller_test.go

```diff
@@ -122,6 +135,154 @@ var _ = Describe("ProjectDevelopmentStream Controller", func() {
 			"projctl_v1beta1_pds_w_existing_comp.yaml",
 		),
 	)
+
+	Context("Status Conditions for edge cases", func() {
```

Comment:
I would expect to see the status condition tests as part of the existing YAML file-based tests we already have, why is that not the case?

Reply:
lets try to demonstrate as many of these cases as we can using the existing example-based system (without major code changes, but with additional files as needed), then we can review what we are left with and consider what it the best system for those tests.

I'm sorry, maybe its subjective but I find these pure code based tests hard to follow.